rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java

```diff
@@ -235,17 +235,18 @@ public void populateRDTDetailsConfirmationPage(WidgetArgs widgetArgs, String dev
         Context context = widgetArgs.getContext();
         if (CovidConstants.FormFields.OTHER_KEY.equals(deviceId)) {
             // reset details when other is selected or device id is blank
-            writeRDTDetailsToWidgets(widgetArgs, context.getString(R.string.unknown_rdt_selected),  "", "");
+            writeRDTDetailsToWidgets(widgetArgs, context.getString(R.string.unknown_rdt_selected),  "", "", "", "");
         } else {
             DeviceDefinitionProcessor deviceDefinitionProcessor = DeviceDefinitionProcessor.getInstance(context, refreshDeviceDefinitionBundle);
             String deviceDetails = getFormattedRDTDetails(widgetArgs.getContext(), deviceDefinitionProcessor.extractManufacturerName(deviceId),
                     deviceDefinitionProcessor.extractDeviceName(deviceId));
             JSONObject deviceConfig = deviceDefinitionProcessor.extractDeviceConfig(deviceId);
-            writeRDTDetailsToWidgets(widgetArgs, deviceDetails, deviceConfig.optString(CovidConstants.FHIRResource.REF_IMG), deviceId);
+            writeRDTDetailsToWidgets(widgetArgs, deviceDetails, deviceConfig.optString(CovidConstants.FHIRResource.REF_IMG), deviceId,
```

Comment:
There's no need to pass `deviceDetails` when it can be constructed from `String manufacturer, String deviceName` which are now being passed to the the `writeRDTDetailsToWidgets` method. 

Instead move `String deviceDetails = getFormattedRDTDetails(widgetArgs.getContext(), deviceDefinitionProcessor.extractManufacturerName(deviceId),
                    deviceDefinitionProcessor.extractDeviceName(deviceId)); ` into writeRDTDetailsToWidgets`

Reply:
done